### PR TITLE
First version of renaming

### LINKED ISSUE #254: Rename root/final to initial/terminal.
- [ x] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [ ] Other?

After a discussion with Dana and Fabian, we decided to change terminology from `root_states` and `final_states` to `initial_states` and `terminal_states`, because these terms are used more frequently in the bio community. It means we have to change this throughout our API.

### PR DESCRIPTION
**Pull request by michalk8**
**IMPORTANT: Please search among the [Pull request](../pulls) before creating one.**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)

## Closes

closes #254 

TODOs:
- [x] check if everthing has been renamed and that there are no artifacts from docrep
- [ ] update tutorials

### CODE REVIEW

File: cellrank/tl/_lineages.py [hunk 1]
```diff
@@ -27,28 +24,27 @@ def lineages(
     """
```
Comment (michalk8) [inline cellrank/tl/_lineages.py:24]: @Marius1311 this docstring needs an update - I think it's still from the time when we've used metastable probabilities from GPCCA as lineaege probabilitiles.

**Comment by michalk8** [comment]
@Marius1311 I've renamed the root -> initial, final -> terminal (in high lvl), could you please check the docs if everything matches?

**Comment by michalk8** [comment]
I've checked this and am merging this, couldn't find any issues - if there are some, they will be fixed later.

File: cellrank/tl/_lineages.py (hunk 1 above)
Comment (Marius1311) [reply cellrank/tl/_lineages.py:24]: I like the way you changed the docstring. I just added a few lines and pushed directly to master.

**Pull request merged**